### PR TITLE
fix: chunk validator rewards

### PR DESCRIPTION
**Pull request by Longarithm**
**UPDATE**: this only fixes `EpochInfoAggregator::merge_common`. Chunk validator threshold is a TODO.

## Old motivation

Here we just introduce `chunk_validator_only_kickout_threshold` = 90 for mainnet. If node only validates chunks and number of endorsed chunks is less than threshold, it should be kicked out. It doesn't seem reasonable to apply this limit to block&chunk producers, see [small discussion](https://near.zulipchat.com/#narrow/stream/295558-core/topic/chunk.20validator.20rewards.26kickouts/near/442898865).

Actually, testing this revealed a serious oversight in stats computation: `EpochInfoAggregator::merge_common` didn't sum up endorsement stats which led to them always being low, which could lead to unpredictable kickouts. This is fixed as well.

Testing:
* `test_validator_kickout_sanity` is extended to check that chunk producer is not kicked out for low endorsement stats, but chunk validator is.
* `test_chunk_validator_kickout` checks that if too many chunks are missing but chunk producer threshold is low, only chunk validator is kicked out.

This required adding new variables to `epoch_config(...)` configuration and removing old ones, which leads to big diff, but reviewing only tests should be enough.
I also want to add a TestLoop test but this change is already big enough.

### CODE REVIEW

**Review by bowenwang1996 — COMMENTED** [review]
> chunk_validator_only_kickout_threshold

Should we lower it to 80? Today the threshold for chunk producer is 80

**Comment by Longarithm** [comment]
> Should we lower it to 80? Today the threshold for chunk producer is 80

Oh, okay. We really should do #11265 to make epoch configs less confusing.

File: core/chain-configs/src/lib.rs [hunk 1]
```diff
@@ -52,6 +52,9 @@ pub const BLOCK_PRODUCER_KICKOUT_THRESHOLD: u8 = 90;
 /// Criterion for kicking out chunk producers.
 pub const CHUNK_PRODUCER_KICKOUT_THRESHOLD: u8 = 90;
```
Comment (tayfunelmas) [inline core/chain-configs/src/lib.rs:53]: should we also make them 80 given that the default is 80 in mainnet?

File: chain/epoch-manager/src/types.rs [hunk 2]
```diff
@@ -291,6 +288,10 @@ impl EpochInfoAggregator {
                             .and_modify(|entry| {
                                 *entry.expected_mut() += stat.expected();
                                 *entry.produced_mut() += stat.produced();
+                                entry.endorsement_stats_mut().expected +=
+                                    stat.endorsement_stats().expected;
+                                entry.endorsement_stats_mut().produced +=
+                                    stat.endorsement_stats().produced;
```
Comment (tayfunelmas) [inline chain/epoch-manager/src/types.rs:294]: oh one more place to fix. thanks for catching this!

File: core/chain-configs/src/lib.rs (hunk 1 above)
Comment (Longarithm) [reply core/chain-configs/src/lib.rs:53]: I don't want to break existing logic. Moreover, there may be tests checking that **at genesis** threshold was at 90 and then it was lowered to 80...

**Review by birchmd — APPROVED** [review]

**Review by tayfunelmas — APPROVED** [review]

**Pull request merged**